operations/src/test/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkServiceTest.java

```diff
@@ -0,0 +1,231 @@
+        service.downloadBulk(request);
+        // the zip file should be empty
+        assertZipFile(0, List.of());
+    }
+
+    @Test
+    public void successTest() throws IOException {
+        when(repoObjLoader.getWorkObject(eq(parentPid))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1, fileObject2));
+        when(aclService.hasAccess(eq(fileObject1Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        when(aclService.hasAccess(eq(fileObject2Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        makeBinaryObject(fileObject1, FILENAME1);
+        makeBinaryObject(fileObject2, FILENAME2);
+
+        service.downloadBulk(request);
+
+        // the zip file should have two entries
+        assertZipFile(2, List.of(FILENAME1, FILENAME2));
+    }
+
+    @Test
+    public void duplicateFilenameTest() throws IOException {
+        when(repoObjLoader.getWorkObject(eq(parentPid))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1, fileObject2));
+        when(aclService.hasAccess(eq(fileObject1Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        when(aclService.hasAccess(eq(fileObject2Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        makeBinaryObject(fileObject1, "filename.txt");
+        makeBinaryObject(fileObject2, "filename.txt");
+
+        service.downloadBulk(request);
+
+        // the zip file should have two entries
+        assertZipFile(2, List.of("filename.txt", "filename(1).txt"));
+    }
+
+    @Test
+    public void multipleFilenamesTest() throws IOException {
+        var fileObject3 = mock(FileObject.class);
+        var fileObject3Pid = makePid();
+        when(fileObject3.getPid()).thenReturn(fileObject3Pid);
+        when(repoObjLoader.getWorkObject(eq(parentPid))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1, fileObject2, fileObject3));
+        when(aclService.hasAccess(eq(fileObject1Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        when(aclService.hasAccess(eq(fileObject2Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        when(aclService.hasAccess(eq(fileObject3Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        makeBinaryObject(fileObject1, "filename.txt");
+        makeBinaryObject(fileObject2, "filename.txt");
+        makeBinaryObject(fileObject3, "flower.txt");
+
+        service.downloadBulk(request);
+
+        // the zip file should have two entries
+        assertZipFile(3, List.of("filename.txt", "filename(1).txt", "flower.txt"));
+    }
+
+    private void assertZipFile(int numberOfEntries, List<String> filenames) throws IOException {
+        var zipFilePath = zipStorageBasePath.toString() + getZipFilename(PARENT_UUID);
```

Comment:
Use `zipStorageBasePath.resolve(getZipFilename(PARENT_UUID))` to construct the path (generally we should prefer using Path over string concatenation for constructing paths since it is safer)